### PR TITLE
Cfg not test macro

### PR DESCRIPTION
**Pull request by Anfid**
## Purpose

Currently it is not possible without unintuitive workarounds to disable parts of code that are intended for testing only, like mocking or stubbing functions, etc. This PR provides the `#[concordium_cfg_test]` macro counterpart that is intended to allow replacing certain functions exclusively for running tests.

## Changes

`#[concordium_cfg_not_test]` macro added, similar to `#[cfg(not(test))]`, but works with `cargo-concordium`

## Checklist

- [X] My code follows the style of this project.
- [X] The code compiles without warnings.
- [X] I have performed a self-review of the changes.
- [X] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [X] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [X] I accept the above linked CLA.

### CODE REVIEW

File: concordium-std-derive/src/lib.rs [hunk 1]
```diff
@@ -1713,3 +1713,24 @@ pub fn concordium_cfg_test(_attr: TokenStream, item: TokenStream) -> TokenStream
     };
     out.into()
 }
+
+/// Sets the cfg exclusively for building, that will be disabled for Wasm and
+/// native tests. Opposite to `#[concordium_cfg_test]` counterpart.
```
Comment (abizjak) [inline concordium-std-derive/src/lib.rs:1718]:
```suggestion
/// If `wasm-test` feature of `concordium-std` is enabled ignore the item. Otherwise this is equivalent to `#[cfg(not(test))]`. Use as a dual to `#[concordium_cfg_test]`.
```

File: concordium-std-derive/src/lib.rs [hunk 2]
```diff
@@ -1713,3 +1713,24 @@ pub fn concordium_cfg_test(_attr: TokenStream, item: TokenStream) -> TokenStream
...
+#[cfg(feature = "wasm-test")]
+#[proc_macro_attribute]
+pub fn concordium_cfg_not_test(_attr: TokenStream, _item: TokenStream) -> TokenStream {
+    TokenStream::new()
+}
+
+/// Sets the cfg exclusively for building, that will be disabled for Wasm and
+/// native tests. Opposite to `#[concordium_cfg_test]` counterpart.
```
Comment (abizjak) [inline concordium-std-derive/src/lib.rs:1726]:
```suggestion
/// If `wasm-test` feature of `concordium-std` is enabled ignore the item. Otherwise this is equivalent to `#[cfg(not(test))]`. Use as a dual to `#[concordium_cfg_test]`.
```
It probably needs to be reformatted though.

File: concordium-std-derive/src/lib.rs [hunk 3]
```diff
@@ -1531,7 +1531,7 @@ fn reject_derive_worker(input: TokenStream) -> syn::Result<TokenStream> {
         }
     };
 
-    let variant_error_conversions = generate_variant_error_conversions(&enum_data, &enum_ident)?;
+    let variant_error_conversions = generate_variant_error_conversions(enum_data, enum_ident)?;
```
Comment (abizjak) [inline concordium-std-derive/src/lib.rs:1534]: I am generally not in favor of making unrelated changes in a PR.

However since this is so small it is OK now.

**Review by abizjak — COMMENTED** [review]
I have a suggestion for improving documentation, otherwise looks good.

File: concordium-std-derive/src/lib.rs (hunk 3 above)
Comment (Anfid) [reply concordium-std-derive/src/lib.rs:1534]: It was required for the point of the checklist "The code compiles without warnings". I was just trying to follow the contributing guide, and calling `clippy --all` gave a couple of warnings. So I decided that might as well fix them in separate commit.

File: concordium-std-derive/src/lib.rs (hunk 1 above)
Comment (Anfid) [reply concordium-std-derive/src/lib.rs]: Not really a fan of the proposed wording. User is usually not aware of `wasm-test` feature and when it gets enabled. I decided that it may be better to elaborate when this feature gets enabled automatically.

File: concordium-std-derive/src/lib.rs (hunk 3 above)
Comment (abizjak) [reply concordium-std-derive/src/lib.rs:1534]: Right, but I think you are probably using a newer version of rust.

The CI uses 1.53, but of course newer versions expand the set of clippy issues, sometimes a bit too eagerly :)

**Review by abizjak — APPROVED** [review]

**Comment by abizjak** [comment]
Looks good @Anfid . Please merge.

**Comment by Anfid** [comment]
I don't have write access to this repository to merge my changes
